File: Backend/src/main/java/com/BankingApplication/service/CustomerService.java
package com.BankingApplication.service;


import java.util.List;

import com.BankingApplication.model.CustomerAccount;

public interface CustomerService {

	public CustomerAccount createCustomerAccount(CustomerAccount account);
	
	public List<CustomerAccount> getAllCustomers();
	
	public CustomerAccount updateCustomerAccount(Long accNo,CustomerAccount updatedAccount);
	
	public String deleteCustomerAccount(Long accNo);
	
	public CustomerAccount searchByAccountNo(Long accNo);
	
	public boolean authenticateCustomer(Long accNo, int accpin);
	
	
	 void deposit(Long accNo, double amount);
	 void transfer(Long fromAccNo, Long toAccNo, double amount);
	 void updatePin(Long accNo, int newPin);
}

File: Frontend-Angular/src/app/services/customer.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class CustomerService {

  apiUrl="http://localhost:8080"

  constructor(private http:HttpClient) { }

  
  getCustomer()
  {
    return this.http.get(`${this.apiUrl}/api/customers/all`)
  }

  addCustomer(customer:any):Observable<any>
  {
    return this.http.post(`${this.apiUrl}/api/customers/create`,customer)
  }

  deleteCustomer(accNo:any)
  {
    return this.http.delete(`${this.apiUrl}/api/customers/delete/${accNo}`,{responseType:'text'})
  }

  updateCustomer(accNo:any,customer:any)
  {
    return this.http.put(`${this.apiUrl}/api/customers/update/${accNo}`,customer);
  }

  getCustomerByAccNo(accNo:any)
  {
    return this.http.get(`${this.apiUrl}/api/customers/getbyId/${accNo}`)
  }
  
  deposit(accNo: number, amount: number) {
    return this.http.post(`${this.apiUrl}/api/customers/deposit`, { accNo, amount });
  }
  

  transfer(fromAccNo: any, toAccNo: any, amount: number): Observable<any> {
    return this.http.post(`${this.apiUrl}/api/customers/transfer`, { fromAccNo, toAccNo, amount });
  }

  updatePin(accNo: any, newPin: number): Observable<any> {
    return this.http.put(`${this.apiUrl}/api/customers/update-pin`, { accNo, newPin });
  }

  customerLogin(accNo: number, accPin: number): Observable<any> {
    return this.http.post(`${this.apiUrl}/api/customers/login`, { accNo, accPin });
  }
}
